### DISCUSSION TITLE
Automatically add all files

### DISCUSSION THREAD

**Discussion by rubiin** [twpayne/chezmoi#3825] [Q&A] [answered]
While editing alot of configs , i sometimes forget to add them to chezmoi and later run into some issues 
so just wanted to ask if there was  a way to automatically add all files managed by chezmoi

**Comment by halostatue** [comment, marked as answer]
Within chezmoi, no and it’s not a feature that is suitable for addition to chezmoi. However #3680 suggests a possible approach that can be used, although it is Linux-specific. #3697 points to `chezmoi managed --include=dirs` that could *seed* files to watch for changes.

There are potential downsides:

- Watching for changes would trigger your watch flow with `chezmoi apply`.
- Using `chezmoi add` to re-add files will destroy any templating you applied. Consider using `chezmoi re-add`, but that will not update templates.

I generally use `chezmoi merge-all` to make sure I have merged all my files prior to `apply`, which allows me to test out changes in realtime but apply periodically for clean updates.
